### PR TITLE
Orca maps

### PR DESCRIPTION
**Pull request by wavid-b**
Orca maps with the merge conflicts fixed

- [x] Fix the wast tests in orca-maps3

### CODE REVIEW

File: src/generator/simple_ast.rs [hunk 1]
```diff
@@ -245,7 +265,19 @@ impl WhammVisitor<()> for SimpleASTBuilder<'_, '_> {
...
+            None => {
+                self.add_probe_to_ast(probe.as_ref(), 0);
+                return;
+            }
+        };
+        for stmt in &stmts.stmts {
+            self.visit_stmt(stmt);
+        }
+        self.add_probe_to_ast(probe.as_ref(), self.curr_num_reports);
+        self.curr_num_reports = 0;
```
Comment (ejrgilbert) [inline src/generator/simple_ast.rs:280]:
```
if probe.body().is_none() {
       self.add_probe_to_ast(probe.as_ref(), 0);
       return;
}
```

**Review by ejrgilbert — CHANGES_REQUESTED** [review]

File: src/generator/instr_generator.rs [hunk 2]
```diff
@@ -22,28 +26,27 @@ fn get_loc_info<'a>(rule: &'a WhammProvider, emitter: &VisitingEmitter) -> Optio
...
-        err: &'e mut ErrorGen,
+        err: &'g mut ErrorGen,
     ) -> Self {
         Self {
             emitter,
             ast,
             err,
             curr_instr_args: vec![],
-            curr_probe_mode: "".to_string(),
+            curr_probe_mode: "____0".to_string(),
```
[the remark this answers is not in the document]
Comment (wavid-b) [reply src/generator/instr_generator.rs:49]: it gets overwriten elsewhere - has to do with alt/before/after

File: src/parser/tests.rs [hunk 3]
```diff
@@ -544,20 +581,15 @@ fn test_global_stmts() {
...
-        BEGIN{
+        wasm::call:alt{
+            (i32, i32) a = (arg0, arg1);
             strcmp((arg0, arg1), "bookings");
         }
-        END {
-            a = 2;
-        }
     "#;
-
```
Comment (ejrgilbert) [inline src/parser/tests.rs:560]: It doesn't look like this is actually testing global statements anymore, maybe add as a new test instead?

File: src/parser/tests.rs [hunk 4]
```diff
@@ -583,8 +615,9 @@ pub fn testing_global_def() {
     setup_logger();
     let mut err = ErrorGen::new("".to_string(), "".to_string(), 0);
     let script = r#"
+        (i32, i32) sample = (1, 2);
         dummy_fn() {
-            a = strcmp((arg0, arg1), "bookings");
+            a = strcmp(sample, "bookings");
             strcmp((arg0, arg1), "bookings");
         }
         i32 i;
```
Comment (ejrgilbert) [inline src/parser/tests.rs:623]: Add as new test instead, more tests is better.

File: src/emitter/map_lib_adapter.rs [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+                Err(e) => Err(e),
+            },
+            _ => Err(Box::new(ErrorGen::get_unexpected_error(
+                true,
+                Some(format!("Non-map with name: {}", name)),
+                None,
+            ))),
+        }
+    }
+    pub fn create_no_meta_map(&mut self, map: DataType) -> Result<(String, i32), Box<WhammError>> {
```
[the remark this answers is not in the document]
Comment (wavid-b) [reply src/emitter/map_lib_adapter.rs:166]: It's a map where we don't care about reporting its key/values. The default map if you dont use "report"

File: src/parser/tests.rs (hunk 4 above)
Comment (wavid-b) [reply src/parser/tests.rs:623]: added to the list of "valid scripts"

**Review by ejrgilbert — APPROVED** [review]

**Pull request merged**